File: replugin-sample/plugin/plugin-demo1/app/src/main/java/com/qihoo360/replugin/sample/demo1/service/PluginDemoService1.java


package com.qihoo360.replugin.sample.demo1.service;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.widget.Toast;

import com.qihoo360.replugin.sample.demo1.support.LogX;
public class PluginDemoService1 extends Service {

    public static final String TAG = "demo.service";

    @Override
    public void onCreate() {
        super.onCreate();

        LogX.logDebug(TAG, "PluginDemoService1 onCreate()");
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        String action = intent.getAction();
        Toast.makeText(this, "PluginDemoService1.action = " + action, Toast.LENGTH_SHORT).show();
        LogX.logDebug(TAG, "PluginDemoService1 onStartCommand()");
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        LogX.logDebug(TAG, "onDestroy()");
    }
}

File: plugin-demo3-kotlin/app/src/main/java/com/qihoo360/replugin/sample/demo3/service/PluginDemoService1.kt


package com.qihoo360.replugin.sample.demo3.service

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.widget.Toast

import com.qihoo360.replugin.sample.demo3.support.LogX

/**
 * @author RePlugin Team
 */
class PluginDemoService1 : Service() {

    override fun onCreate() {
        super.onCreate()

        LogX.logDebug(TAG, "onCreate()")
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        val action = intent.action
        Toast.makeText(this, "PluginDemoService1.action = " + action, Toast.LENGTH_SHORT).show()
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onDestroy() {
        super.onDestroy()
        LogX.logDebug(TAG, "onDestroy()")
    }

    companion object {

        val TAG = "demo.service"
    }
}
